poetry/installation/executor.py

```diff
@@ -0,0 +1,416 @@
+            self._install_git(package)
+
+            return
+
+        archive = self._download(operation)
+        operation_message = self.get_operation_message(operation)
+        message = "  <fg=blue;options=bold>•</> {message}: <info>Installing...</info>".format(
+            message=operation_message,
+        )
+        if not self._io.output.supports_ansi() or self._io.is_debug():
+            self._io.write_line(message)
+        else:
+            self._io.overwrite(message)
+
+        args = ["install", "--no-deps", str(archive)]
+        if operation.job_type == "update":
+            args.insert(2, "-U")
+
+        self.run(*args)
+
+        message = "  <fg=green;options=bold>✓</> {message}".format(
+            message=operation_message,
+        )
+        if not self._io.output.supports_ansi() or self._io.is_debug():
+            self._io.write_line(message)
+        else:
+            self._io.overwrite(message)
+            self._io.write_line("")
+
+    def _update(self, operation):
+        return self._install(operation)
+
+    def _remove(self, operation):
+        package = operation.package
+
+        # If we have a VCS package, remove its source directory
+        if package.source_type == "git":
+            src_dir = self._env.path / "src" / package.name
+            if src_dir.exists():
+                safe_rmtree(str(src_dir))
+
+        try:
+            self.run("uninstall", package.name, "-y")
+        except CalledProcessError as e:
+            if "not installed" in str(e):
+                return
+
+            raise
+
+    def _install_directory(self, package, from_vcs=False):
+        from poetry.factory import Factory
+        from poetry.core.masonry.builder import SdistBuilder
+        from poetry.utils._compat import decode
+        from poetry.utils.env import NullEnv
+        from poetry.utils.toml_file import TomlFile
+
+        if not from_vcs:
+            message = "  - <c1>{}</c1> (<c2>{}</c2>): <fg=blue;options=bold>•</> Installing".format(
+                package.name, package.full_pretty_version
+            )
+            self._io.write_line(message)
+
+        if package.root_dir:
+            req = os.path.join(package.root_dir, package.source_url)
+        else:
+            req = os.path.realpath(package.source_url)
+
+        args = ["install", "--no-deps", "-U"]
+
+        pyproject = TomlFile(os.path.join(req, "pyproject.toml"))
+
+        has_poetry = False
+        has_build_system = False
+        if pyproject.exists():
+            pyproject_content = pyproject.read()
+            has_poetry = (
+                "tool" in pyproject_content and "poetry" in pyproject_content["tool"]
+            )
+            # Even if there is a build system specified
+            # pip as of right now does not support it fully
+            # TODO: Check for pip version when proper PEP-517 support lands
+            # has_build_system = ("build-system" in pyproject_content)
```

Comment:
I'm not sure I follow -- could you elaborate on what this means?

Reply:
This is an old comment that is no longer true and the appropriate changes have already be done on Poetry's side